ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml

```diff
@@ -176,93 +176,108 @@ Popup {
-                    activityCenterStore: root.activityCenterStore
-                    notification: model
-                    onCloseActivityCenter: root.close()
-                }
-            }
-            DelegateChoice {
-                roleValue: Constants.activityCenterNotificationTypeCommunityMembershipRequest
-
-                ActivityNotificationCommunityMembershipRequest {
-                    width: listView.availableWidth
-                    filteredIndex: index
-                    store: root.store
-                    activityCenterStore: root.activityCenterStore
-                    notification: model
-                    onCloseActivityCenter: root.close()
-                }
-            }
-            DelegateChoice {
-                roleValue: Constants.activityCenterNotificationTypeCommunityRequest
-
-                ActivityNotificationCommunityRequest {
-                    width: listView.availableWidth
-                    filteredIndex: index
-                    store: root.store
-                    activityCenterStore: root.activityCenterStore
-                    notification: model
-                    onCloseActivityCenter: root.close()
-                }
-            }
-            DelegateChoice {
-                roleValue: Constants.activityCenterNotificationTypeCommunityKicked
-
-                ActivityNotificationCommunityKicked {
-                    width: listView.availableWidth
-                    filteredIndex: index
-                    store: root.store
-                    activityCenterStore: root.activityCenterStore
-                    notification: model
-                    onCloseActivityCenter: root.close()
+        delegate: Loader {
+            width: listView.availableWidth
+
+            property int filteredIndex: index
+            property var notification: model
+
+            sourceComponent: {
+                switch (model.notificationType) {
+                    case Constants.activityCenterNotificationTypeMention:
+                        return mentionNotificationComponent
+                    case Constants.activityCenterNotificationTypeReply:
+                        return replyNotificationComponent
+                    case Constants.activityCenterNotificationTypeContactRequest:
+                        return contactRequestNotificationComponent
+                    case Constants.activityCenterNotificationTypeCommunityInvitation:
+                        return communityInvitationNotificationComponent
+                    case Constants.activityCenterNotificationTypeCommunityMembershipRequest:
+                        return membershipRequestNotificationComponent
+                    case Constants.activityCenterNotificationTypeCommunityRequest:
+                        return communityRequestNotificationComponent
+                    case Constants.activityCenterNotificationTypeCommunityKicked:
+                        return communityKickedNotificationComponent
```

Comment:
Add a `default` case here with some warning for unhandled types

Reply:
fixed